src/lib/components/controls/PointerLockControls.svelte

```diff
@@ -0,0 +1,58 @@
+<script>
+	import { setup } from '../../utils/context.js';
+	import { PointerLockControls } from 'three/examples/jsm/controls/PointerLockControls.js';
+	import { createEventDispatcher } from 'svelte';
+
+	export let maxPolarAngle = Math.PI;
+	export let minPolarAngle = 0;
+
+	let isLocked = false;
+
+	const { root } = setup();
+	const dispatch = createEventDispatcher();
+
+	/** @type {PointerLockControls} */
+	let controls;
+
+	root.controls.set((camera, canvas) => {
+
+		controls = new PointerLockControls(camera, canvas);
+
+		controls.addEventListener('start', (e) => {
+			dispatch('start', e);
+		});
+
+		controls.addEventListener('end', (e) => {
+			dispatch('end', e);
+		});
+
+		controls.addEventListener('lock', (e) => {
+			isLocked = true;
+			dispatch('lock', e);
+		});
+
+		controls.addEventListener('unlock', (e) => {
+			isLocked = false;
+			dispatch('unlock', e);
+		});
+
+		// allows rotating the camera
+		controls.addEventListener('change', (e) => {
+			dispatch('change', e);
+
+			root.invalidate();
+		});
+
+		canvas.addEventListener('click', () => {
```

Comment:
I think we should remove all event listeners onDestroy https://svelte.dev/docs#onDestroy and also reset camera properties if needed

Reply:
I thought it's not necessary, because the event listenders should get removed automatically as soon as the component is unmounted, because the object, on which the event listenters have been added to, doesn't exist any more.
If they really stay there I totally agree, that they should be removed.

In what case would you reset the camera properties? Either I am too sleepy or I haven't changed any camera properties in the current pull request?